src/main/python/main/ayab/engine/engine.py

```diff
@@ -166,37 +168,51 @@ def run(self, operation: Operation) -> None:
         self.config.portname = self.__read_portname()
         self.control.start(self.pattern, self.config, operation)
 
-        while True:
-            # continue operating
-            # typically each step involves some communication with the device
-            output = self.control.operate(operation)
-            if output != self.control.notification:
-                self.__feedback.handle(output)
-                self.control.notification = output
-            if operation == Operation.KNIT:
-                self.__handle_status()
-            if self.__canceled or self.control.state == State.FINISHED:
-                break
-
-        self.control.stop()
+        keep_mode = None
+        if self.preferences.value("keep_awake"):
+            self.__logger.debug("Keeping screen awake activated")
+            # Activate the keep mode only if the preference is activated
+            keep_mode = keep.presenting(on_fail="pass")
+            keep_mode.__enter__()
+
+        try:
```

Comment:
I still think it shouldn't be a preference, but if it has to be one, at least it can be made a bit more palatable with `nullcontext` that is designed for such cases:

```suggestion
        from contextlib import nullcontext # at the top
        …
        with keep.presenting(on_fail="pass") if self.preferences.value("keep_awake") else nullcontext():
```

Then you can get rid of the `try…finally` block.

Reply:
If it is okay for others, I'd prefer removing the settings ...
But if not yes that's a nicer solution.

Reply:
What was the conclusion of this discussion? Having it as a preference yes or no?
@X-sam  @jonathanperret

Reply:
My vote is still for no preference.

Reply:
I would also vote for no preference as I don't think someone would want his computer to sleep while knitting.

Reply:
Ok, then let's go without a preference for now. If somebody finds it really annoying we can add the preference setting later on.
@mathPi would you mind adjusting your PR?

Reply:
It is already done ;)